### PR TITLE
Elaborate on connections between Logic and lists

### PR DESCRIPTION
**Pull request by Bodigrim**
CC @kquick

### CODE REVIEW

File: Control/Monad/Logic.hs [hunk 1]
```diff
@@ -176,11 +196,14 @@ logic f = LogicT $ \k -> Identity .
...
 --
+-- Since 'Logic' is isomorphic to a list, 'observe' is 'head' in disguise.
+--
 observe :: Logic a -> a
 observe lt = runIdentity $ unLogicT lt (const . pure) (error "No answer.")
 
 -------------------------------------------------------------------------
--- | Extracts all results from a 'Logic' computation.
+-- | Extracts all results from a 'Logic' computation, witnessing
+-- a half of isomorphism between 'Logic' and lists.
```
Comment (kquick) [inline Control/Monad/Logic.hs:206]: For clarity in description v.s. insight, I'd recommend removing the ", witnessing..." here and instead, after the example code, write "The 'observeAll' function witnesses the other half of the isomorphism between 'Logic' and lists, where the first half is described in the 'runLogic' description above."

File: Control/Monad/Logic.hs [hunk 2]
```diff
@@ -176,11 +196,14 @@ logic f = LogicT $ \k -> Identity .
...
 --
 -- >>> observe (pure 5 <|> empty <|> empty <|> pure 3 <|> empty)
```
Comment (kquick) [inline Control/Monad/Logic.hs:208]: Oops!  Not something exactly pertinent to this PR, but it looks like this should have been `observeAll` in this example.

**Review by kquick — APPROVED** [review]
Good info, and especially the link to the Boehm-Berarducci background.

I did have a few suggestions for distinguishing the isomorphism from the general descriptive information.  My thought is that  providing the background and intuition separately from the base description will help people trying to understand `Logic`.  That said, these are soft suggestions so feel free to ignore or adapt them as you'd like.

There is one error in one of the examples; not something introduced in this PR (I think it was from mine), but it's probably worth fixing it in this PR.

**Comment by Bodigrim** [comment]
@kquick thanks for review, updated.

**Pull request merged**